solr/src/main/java/org/apache/metamodel/solr/SolrDataContext.java

```diff
@@ -0,0 +1,396 @@
+            HttpGet request = new HttpGet(url);
+            HttpResponse response = httpclient.execute(request);
+
+            InputStreamReader inputStream = new InputStreamReader(response
+                    .getEntity().getContent());
+
+            ObjectReader reader = new ObjectMapper().reader(HashMap.class);
+            Map<String, Object> outerMap = reader.readValue(inputStream);
+
+            Map<String, Object> schemaMap = null;
+
+            if (outerMap != null) {
+                schemaMap = (Map) outerMap.get("schema");
+            }
+
+            if (schemaMap != null) {
+                List<Map<String, String>> fieldMapList = (List) schemaMap
+                        .get("fields");
+                List<String> columnNames = new ArrayList<String>();
+                List<ColumnType> columnTypes = new ArrayList<ColumnType>();
+
+                for (int i = 0; i < fieldMapList.size(); i++) {
+                    Map<String, String> fieldMap = fieldMapList.get(i);
+
+                    columnNames.add(fieldMap.get("name"));
+                    columnTypes.add(ColumnType.STRING);
+                }
+
+                String[] columnNamesArr = new String[columnNames.size()];
+                columnNamesArr = columnNames.toArray(columnNamesArr);
+
+                ColumnType[] columnTypesArr = new ColumnType[columnTypes.size()];
+                columnTypesArr = columnTypes.toArray(columnTypesArr);
+
+                tableDef = new SimpleTableDef(indexName, columnNamesArr,
+                        columnTypesArr);
+            }
+        } catch (IOException e) {
+            logger.error("Failed to parse schema", "", e);
+        }
+
+        return tableDef;
+    }
+
+    @Override
+    protected Schema getMainSchema() throws MetaModelException {
+        try {
+            final MutableSchema theSchema = new MutableSchema(
+                    getMainSchemaName());
+
+            final MutableTable table = tableDef.toTable().setSchema(theSchema);
+            theSchema.addTable(table);
+
+            return theSchema;
+        } catch (Exception e) {
+            throw new MetaModelException("Schema retrieval failed " + e);
+        }
+    }
+
+    @Override
+    protected Number executeCountQuery(Table table,
+            List<FilterItem> whereItems, boolean functionApproximationAllowed) {
+        String queryStr = "*:*";
+
+        if (whereItems != null) {
+            for (int i = 0; i < whereItems.size(); i++) {
+                FilterItem whereItem = whereItems.get(i);
+                queryStr = whereItem.toString();
+            }
+        }
+
+        QueryResponse response = selectRows(table, null, queryStr, 0, 0);
+        return response.getResults().getNumFound();
+    }
+
+    @Override
+    public DataSet executeQuery(Query q) throws MetaModelException {
+        String queryStr = "";
+        QueryResponse response = null;
+
+        Schema schema = getMainSchema();
```

Comment:
The schema is referred to in the Query, it might not be the main schema - it might be the information schema.